packages/components/src/internal/actions.ts

```diff
@@ -2013,143 +1743,111 @@ async function prepareInsertRowDataFromBulkForm(
+
+    if (paste.success) {
+        const byColumnValues = getPasteValuesByColumn(paste);
+        // prior to load, ensure lookup column stores are loaded
+        const columnLoaders: any[] = queryInfo.getInsertColumns().reduce((arr, column, index) => {
+            if (column.isPublicLookup()) {
+                const filteredLookup = getColumnFilteredLookup(column, columnMetadata);
+                if (
+                    index >= paste.coordinates.colMin &&
+                    index <= paste.coordinates.colMax &&
+                    byColumnValues.get(index - paste.coordinates.colMin).size > 0
+                ) {
+                    arr.push(
+                        findLookupValues(
+                            column,
+                            undefined,
+                            filteredLookup
+                                ? filteredLookup.toArray()
+                                : byColumnValues.get(index - paste.coordinates.colMin).toArray()
+                        )
+                    );
+                } else if (filteredLookup) {
+                    arr.push(findLookupValues(column, undefined, filteredLookup.toArray()));
                 }
-                return arr;
-            }, []);
-
-            Promise.all(columnLoaders)
-                .then(results => {
-                    const descriptorMap = {};
-                    results.forEach(result => {
-                        const { column, descriptors } = result;
-                        descriptorMap[column.lookupKey] = descriptors;
-                    });
-                    return pasteCellLoad(
-                        model,
-                        gridModel,
-                        paste,
-                        descriptorMap,
-                        columnMetadata,
-                        readonlyRows,
-                        lockRowCount
-                    ).then(payload => {
-                        model = updateEditorModel(model, {
-                            cellMessages: payload.cellMessages,
-                            cellValues: payload.cellValues,
-                            selectionCells: payload.selectionCells,
-                        });
-
-                        model = endPaste(model);
-                    });
-                })
-                .then(() => {
-                    if (onComplete) {
-                        onComplete();
-                    }
-                })
-                .catch(reason => {
-                    console.error(reason);
-                });
-        } else {
-            const cellKey = genCellKey(colIdx, rowIdx);
-            model = updateEditorModel(model, {
-                cellMessages: model.cellMessages.set(cellKey, { message: paste.message } as CellMessage),
-            });
-        }
+            }
+            return arr;
+        }, []);
+
+        const results = await Promise.all(columnLoaders);
```

Comment:
It seems to be working for me, but maybe my testing scenarios are too limited. Do you have a particular test case I can test against?

Reply:
Unfortunately, I don't remember the specific failure case, but I believe I encountered the problem while doing the conversion of lookups to use QuerySelect.  It might manifest if you are using filters for the lookups in the grid, which happens, at least, in the grid shown when adding samples to storage (the samples already in storage or with locked or consumed status should be filtered out).

@cnathe I think we looked at that together at the time.  Do you happen to remember more particulars?

Reply:
All this being said, I've just encountered a different place where there's an `await` with a `Promise.all` (in `ParentEntityEditPanel`). It would still be good to test the case of the Add to Storage editable grid (with samples already in storage), but the problem I had was likely because of something else.

Reply:
I can't remember the particulars, but it was pretty clear that the Promise.all wasn't waiting, so I agree that it was likely a usage problem and not a general `await Promise.all` problem.

Reply:
Is this scenario covered by any of our FM Selenium tests?

Reply:
Yes, in `FMAddSamplesToLocationsTest` we check this.